src/methods.jl

```diff
@@ -155,4 +155,6 @@ Base.literal_pow(::typeof(^), x::Symbolic{<:Number}, ::Val{p}) where {p} = Base.
 # Array-like operations
 Base.size(x::Symbolic{<:Number}) = ()
 Base.length(x::Symbolic{<:Number}) = 1
-Base.ndims(x::Symbolic{<:Number}) = 0
+Base.ndims(x::Symbolic{T}) where {T} = Base.ndims(T)
+Base.ndims(::Type{<:Symbolic{T}}) where {T} = Base.ndims(T)
+Base.broadcastable(x::Symbolic{T}) where {T} = Ref(x)
```

Comment:
Maybe should restrict this to `T<:Number`?

Reply:
Actually, I think just `T` should be correct here as a reasonable default. Any more specific implementation could still be defined, but likely would look the same for any "scalar" type.
Otherwise, the test for the custom `SquareDummy` type would also fail.

Reply:
I think it is correct only if Base had `Base.broadcastable(x::Any) = Ref(x)`. But it goes to some lengths to avoid defining it.
```julia
julia> @which Base.broadcastable("hi")
broadcastable(x::Union{Missing, Nothing, UndefInitializer, AbstractPattern, AbstractString, Function, Symbol, Pair, Val, Ptr, RoundingMode}) in Base.Broadcast at broadcast.jl:678
```